### PR TITLE
Provide runtime statistics to statistics.py

### PR DESCRIPTION
**Pull request by boudewijn-tribler**
This is a requirement for https://github.com/Tribler/tribler/pull/294

### CODE REVIEW

**Comment by NielsZeilemaker** [comment]
Seems fine, maybe add the runtime statistics to some other cpu intensive tasks as well.
Creating signatures? Time it takes to create a message of a given type?

**Comment by boudewijn-tribler** [comment]
Wrapping the Conversion.encode_message and Conversion.decode_message methods will give us creating and verifying of signatures and the byte packing stuff per message message.  I'll update the code so you can have a look.

**Comment by NielsZeilemaker** [comment]
Maybe wrap the encode/decode_member_authentication and encode/decode_double_member_authentications to get a more clear understanding of just the signing part of messages.

**Comment by boudewijn-tribler** [comment]
I did and all it shows is that approximately 90% of the decode is with the crypto verify and approximately 80% of the encode is in the signing part.

Of course these percentages won't be the same for every message, but I remember measuring this about a year ago with the same results.  So before I push this commit, do you still think this information is interesting enough to have it measured all the time?

**Comment by NielsZeilemaker** [comment]
Yes, because if we measure it we can see if these assumptions continue to hold.

**Comment by boudewijn-tribler** [comment]
Segmentation fault occurring after the tests run successfully...  Elric, can you take a look at this please?

File: decorator.py [hunk 1]
```diff
@@ -71,64 +73,101 @@ def helper(*args, **kargs):
...
-           foo(2)
-
-        After running the above example, the statistics will show that:
-        - 'foo bar=1 moo=milk' was called once
-        - 'foo bar=2 moo=milk' was called twice
-        """
-        def helper(func):
-            return func
-        return helper
+    Runtime statistics will only be collected when sys.argv contains '--runtime-statistics'.
```
Comment (NielsZeilemaker) [inline decorator.py]: Not true

**Pull request merged**